### ISSUE TITLE
Sniff for use of `filemtime()`

### ISSUE THREAD

**Issue by terriann** [Automattic/VIP-Coding-Standards#684]
## What problem would the enhancement address for VIP?

Because of the container-based infrastructure, VIP does not recommend using the function `filemtime()` to populate a value for versioning; each time a container is created, it would result in a different value. It would be great to warn developers of this using the coding standards and have the bot notify in pull requests.

We provide this information in the docs for [file concatenation and minification](https://docs.wpvip.com/technical-references/vip-platform/file-concatenation-and-minification/), but a CS sniff would be a better way to widely share that information.

## Describe the solution you'd like

A warning on the use of `filemtime()` due to the container-based infrastructure.

Other PHP functions return a time based on when an action was taken against a file in the file system.  It would be worth considering if this sniff should apply to these too:

* https://www.php.net/manual/en/function.filectime.php
* https://www.php.net/manual/en/function.filemtime.php
* https://www.php.net/manual/en/function.getlastmod.php
* https://www.php.net/manual/en/function.fileatime.php

## What code should be reported as a violation?

```php
function add_js_scripts(){
    wp_register_script(
        'script-name',
        get_template_directory_uri() . '/js/script.js', 
        array(),
        filemtime( get_template_directory() . '/js/script.js' ), 
        false
    ); 
    wp_enqueue_script( 'script-name' );
}
add_action( 'wp_enqueue_scripts', 'add_js_scripts' );
```

```php
function add_js_scripts(){

    $ver = filemtime( get_template_directory() . '/js/script.js' );

    wp_register_script(
        'script-name',
        get_template_directory_uri() . '/js/script.js', 
        array(),
        $ver, 
        false
    ); 
    wp_enqueue_script( 'script-name' );
}
add_action( 'wp_enqueue_scripts', 'add_js_scripts' );
```

## What code should *not* be reported as a violation?

None. Even using `filemtime()` outside of the context of enqueuing could be problematic.

**Comment by GaryJones** [comment]
Given the response in p9zhOE-3lX-p2 (internal), it seems the documentation is incorrect, and `filemtime()` does not change per autoscaling pod; only new deploys update the `filemtime()` and others.
